scmrepo/fs.py

```diff
@@ -20,6 +20,124 @@
     from scmrepo.git.objects import GitTrie
 
 
+class Path:
```

Comment:
Temporary copy from dvc/fs/path.py until we get it into fsspec. Useful for handling chdirs.